libs/dh/wholesale/feature-settlement-reports/src/lib/tabs/dh-wholesale-settlements-reports-tabs-balance.component.ts

```diff
@@ -208,28 +211,30 @@ export class DhWholesaleSettlementsReportsTabsBalanceComponent
   }
 
   downloadClicked(gridAreas: settlementReportsTableColumns[]) {
+    const fileOptions = { name: 'SettlementReport.zip', type: 'application/zip' };
+
     this.toastService.open({
       type: 'loading',
       message: this.transloco.translate('wholesale.settlementReports.downloadStart'),
     });
-    this.settlementReportStore.download(
-      {
-        gridAreas: gridAreas.map((g) => g.id),
-        processType: WholesaleProcessType.BalanceFixing,
-        periodStart: this.searchForm.controls.period.value?.start ?? '',
-        periodEnd: this.searchForm.controls.period.value?.end ?? '',
-        energySupplier: this.searchForm.controls.actor.value ?? undefined,
-        locale: this.transloco.translate('selectedLanguageIso'),
-      },
-      () => {
-        this.toastService.open({
-          type: 'danger',
-          message: this.transloco.translate('wholesale.settlementReports.downloadFailed'),
-        });
-      },
-      () => {
-        this.toastService.dismiss();
-      }
-    );
+
+    this.httpClient
+      .v1WholesaleSettlementReportDownloadGet(
+        gridAreas.map((g) => g.id),
+        WholesaleProcessType.BalanceFixing,
+        this.searchForm.controls.period.value?.start ?? '',
```

Comment:
Question: Should we keep the named parameters?

Reply:
Would love to, but I'm using the generated `httpClient` directly here. We could consider using the `useObjectParameters` setting for the openapi generator (https://openapi-generator.tech/docs/generators/typescript/#config-options). Perhaps in a separate PR, since this will require a fair amount of refactoring? 🙂